ndindex/slice.py

```diff
@@ -373,7 +491,10 @@ def as_subindex(self, index):
             return Tuple(self).as_subindex(index)
 
         if isinstance(index, Integer):
-            s = self.as_subindex(Slice(index.args[0], index.args[0] + 1))
+            if index.args[0] == -1:
```

Comment:
What is this condition testing?

Reply:
If the integer index is equal to -1.

Reply:
So it doesn't need to test for any negative int?

Reply:
-1 is the only integer where slice(i, i+1) doesn't index one element.

Reply:
Ahh I see